passkeys-backend/assets/index.html

```diff
@@ -0,0 +1,336 @@
+        });
+
+        const checkAvalibility = () => {
+            const username = usernameElement.value
+            if(username) {
+                continueButton.classList.add("enable");
+                continueButton.disabled = false;
+            } else {
+                continueButton.classList.remove("enable");
+                continueButton.disabled = true;
+            }
+        }
+
+        function ArrayBufferToBase64(buffer) {
+            // Convert ArrayBuffer to base64 string
+            const binary = String.fromCharCode.apply(null, new Uint8Array(buffer));
+            let base64String = btoa(binary);
+
+            // Replace characters not allowed in base64url with their corresponding URL-safe alternatives
+            base64String = base64String.replace(/\+/g, '-').replace(/\//g, '_').replace(/=/g, '');
+
+            return base64String;
+        }
+
+        const strToArrayBuffer = (str) => {
+            const decodedString = atob(str);
+
+            const buffer = new Uint8Array(decodedString.length);
+            for (let i = 0; i < buffer.length; i++) {
+                buffer[i] = decodedString.charCodeAt(i);
+            }
+            return buffer;
+        }
+
+        function stringToArrayBuffer(str) {
+            const encoder = new TextEncoder();
+            const uint8Array = encoder.encode(str);
+            return uint8Array.buffer;
+        }
+
+        const login = () => {
+            const authenticationCard = document.getElementById("container");
+            const passkeyCard = document.getElementById("modal");
+            authenticationCard.classList.add("invisible");
+            passkeyCard.classList.remove("invisible");
+        }
+
+        const signIn = async () => {
+            try {
+                const response = await fetch(`./authentication/start`);
+                const responseJSON = await response.json();
+
+
+                window.webauthnJSON.get(responseJSON)
+                    .then(async (publicKeyCredential) => {
+                        const { id, rawId, response, type, authenticatorAttachment } = publicKeyCredential
+                        const { authenticatorData, clientDataJSON, signature, userHandle } = response
+
+                        const authentication = await fetch('./authentication/verification', {
+                            method: "POST",
+                            headers: {
+                                "Content-Type": "application/json"
+                            },
+                            body: JSON.stringify({
+                                id: id,
+                                rawId: rawId,
+                                clientDataJson: clientDataJSON,
+                                authenticatorData: authenticatorData,
+                                signature: signature,
+                                userHandle: userHandle
+                            })
+                        });
```

Comment:
Same comment as below but I think we can skip the construction here and use the publicKeyCredential directly in the JSON without deconstructing/reconstructing it.

Reply:
Same situation as below here.